### ISSUE TITLE
wxPersistenceManager collides with static user wxConfig when using executable name

### ISSUE THREAD

**Issue by pieter3d** [wxWidgets/wxWidgets#24169]
wxConfig creates $HOME/.app_name, from the constructor `wxConfig("app_name")`, on my Ubuntu machine.
I create this pointer in a static instance:
```c++
class Workspace {
 public:
  static Workspace &Get() {
    static Workspace w;
    return w;
  }
  wxConfig* GetConfig() const { return cfg_.get(); }
 private:
  Workspace() {
    cfg_ = std::make_unique<wxConfig>("floorplay_cfg");
  }
 std::unique_ptr<wxConfig> cfg_;
};


```
When the application exits, the config data is flushed to the config file, as expected, due to the `std::unique_ptr` destructor running when the `Workspace` static instance is destructed.

However, if there is anywhere a `wxPersistenceManager`, its data gets wiped out since:

- It appears that the `wxPersistenceManager` uses the programs binary file name to come up with a config file name (e.g. "app_name" in this case), and this same string passed to the `wxConfig` constructor.
- The GUI destructs earlier than the static `wxConfig` object.
- I can see this happening when running the program with `strace`, I see two writes to the config file, preceded by an `unlink` which wipes out earlier writes.

Right now the only workaround in this case is to pick a different name for the application config, and as a result there are at least two unique config files. That's pretty messy.

Possible solutions

- Allow user to set the config object name for the `wxPersistenceManager`
- Allow user to provide a config object.

### Platform and version information

- wxWidgets version  you use: v3.0.5
- wxWidgets port  you use: wxGTK
- OS  and its version: Ubuntu 22.04.3 LTS

**Comment by vadz** [comment]
There are a couple of suspicious things going on here:

1. You generally can't use any wx objects before the library initialization or after the library shutdown, which means that using any of them in static objects ctors/dtors is officially unsupported.
1. `wxConfig` is supposed to be used as a singleton, i.e. you should use `wxConfig::Get()` and share the same config object for the entire application. And if you really absolutely need to use another config object, you should use a different file for it to avoid conflicts with the global one, which is used by other things and not just `wxPersistenceManager`.

Practically speaking, you can use `wxApp::SetAppName()` to change the name used by the global config, but this would still result in 2 config files being used. To avoid this, just use `wxConfig::Get()` in your workspace.

To summarize, I don't really think there is any bug here, so closing.
